src/main/java/jenkins/plugins/git/traits/PruneStaleBranchTrait.java

```diff
@@ -44,6 +46,15 @@ public PruneStaleBranchTrait() {
         super(new PruneStaleBranch());
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void decorateContext(SCMSourceContext<?, ?> context) {
+        GitSCMSourceContext<?,?> ctx = (GitSCMSourceContext<?, ?>) context;
```

Comment:
This line seems to cause a class cast exception as reported in [JENKINS-57683](https://issues.jenkins-ci.org/browse/JENKINS-57683).

Reply:
Should probably either have the descriptor limit its applicability or check before casting

Reply:
See #696 for the test that shows the failure and the proposed change that fixes the test.